torch/_inductor/kernel/mm.py

```diff
@@ -164,9 +164,18 @@ def tuned_int_mm(mat1, mat2, *, layout=None):
     choices = (
         [aten__int_mm.bind((mat1, mat2), layout)] if use_aten_gemm_kernels() else []
     )
-    if m * n != 0 and use_triton_template(layout, enable_int32=True):
-        # TODO: Re-enable eager mode implementation once cuBLAS is fixed
+
+    # TODO: Re-enable eager mode implementation once cuBLAS is fixed
+    if m * n != 0 and (
+        use_cutlass_template(layout) or use_triton_template(layout, enable_int32=True)
+    ):
         choices = []
+
+    if m * n != 0 and use_cutlass_template(layout):
+        CUTLASSGemmTemplate.add_cutlass_gemm_choices(
+            choices, layout, [mat1, mat2], fuseable=True, non_fuseable=True
```

Comment:
Why is both fuseable and non_fuseable True? I'm mostly just confused by the names here.

Reply:
It's that an ordinary `_int_mm()` operation is found by parser, so no fusion needed here.  On the other side, CUTLASS code generator differentiates between creating fuseable or non-fuseable MM kernels; fusion is not needed in this particular case, so auto-tuner could (and should) try both fuseable and non-fuseable kernels, in order to find the best performing one.  (I was confused by names on the first sight too, but they actually make sense.)